sklearn/preprocessing/_polynomial.py

```diff
@@ -572,8 +573,14 @@ class SplineTransformer(TransformerMixin, BaseEstimator):
         i.e. a column of ones. It acts as an intercept term in a linear models.
 
     order : {'C', 'F'}, default='C'
-        Order of output array. 'F' order is faster to compute, but may slow
-        down subsequent estimators.
+        Order of output array in the dense case. `'F'` order is faster to compute, but
+        may slow down subsequent estimators.
+
+    sparse : bool, default=False
```

Comment:
I think `sparse`, can get confused with "SplineTransformer accepts sparse as input".

Do you think `sparse_output` is better?

Reply:
I think we should be consistent with `OneHotEncoder(sparse=True)`.

Reply:
I also prefer the more explicit `sparse_output`. Furthermore it would be consistent with:

- `RandomTreesEmbedding`
- `LabelBinarizer`
- `MultiLabelBinarizer`

I would also be in favor of aligning `OneHotEncoder` to that naming convention in a follow-up PR (with a deprecation cycle).